frontend_tests/node_tests/stream_data.js

```diff
@@ -201,6 +201,23 @@ zrequire('stream_data');
     stream_data.update_subscribers_count(sub);
     assert.equal(sub.subscriber_count, 0);
 
+    // verify that checking subscription with bad email is a noop
+    var bad_email = 'notbrutus@zulip.org';
+    global.blueslip.error = function (msg) {
+        assert.equal(msg, "Unknown email for get_user_id: " + bad_email);
+    };
+    global.blueslip.warn = function (msg) {
+        assert.equal(msg, "Bad email passed to user_is_subscribed: " + bad_email);
+    };
+    assert(!stream_data.user_is_subscribed('Rome', bad_email));
+
+    // Verify noop for bad stream when removing subscriber
+    var bad_stream = 'UNKNOWN';
+    global.blueslip.warn = function (msg) {
+        assert.equal(msg, "We got a remove_subscriber call for a non-existent stream " + bad_stream);
+    };
+    ok = stream_data.remove_subscriber('UNKNOWN', brutus.user_id);
```

Comment:
Same as above, the test should also check that when you pass in a valid stream, it does delete the subscription.

Reply:
Also the case where the stream exists but the user isn't subscribed.